wgpu-types/src/lib.rs

```diff
@@ -14,6 +14,48 @@ use serde::{Deserialize, Serialize};
 use std::hash::{Hash, Hasher};
 use std::{num::NonZeroU32, ops::Range};
 
+// Use this macro instead of the one provided by the bitflags_serde_shim crate
+// because the latter produces an error when deserializing bits that are not
+// specified in the bitflags, while we want deserialization to succeed and
+// and unspecified bits to lead to errors handled in wgpu-core.
+// Note that plainly deriving Serialize and Deserialized would have a similar
```

Comment:
Any reason we can't just do that?

Reply:
I am not sure. [The commit that introduced the bitflags serde shim](https://github.com/gfx-rs/wgpu/commit/d5ba0b439d5722980ab958842fa0fe23112548b0) was about deno/cts and doesn't explain why. From a few questions on the chat, my understanding is that it gives you nicer looking serialization by avoiding to wrap all bitflags in single-member structs.

I don't mind removing the custom serialization code, although I'd keep the macro in order to have `contains_invalid_bits` so the cost of having the manual serialize/deserialize written once in the macro doesn't seem particularly bad to me. Let me know what you prefer.

Reply:
The current way SGTM then.